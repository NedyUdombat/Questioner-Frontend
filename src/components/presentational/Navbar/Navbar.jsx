import React from 'react';
import { Icon } from 'semantic-ui-react';

// styles
import './Navbar.scss';
import '../Button/Button.scss';

const Navbar = () => {
  return (
    <div>
      <div className="nav">
        <input type="checkbox" id="nav-check" />
        <div className="nav-header">
          <p className="nav-title">Questioner</p>
        </div>
        <div className="nav-btn">
          <label htmlFor="nav-check">
            <Icon name="bars" color="grey" size="big" />
          </label>
        </div>

        <div className="nav-links">
          <a href="#" className="btn btn-transparent">
            MEETUPS
          </a>
          <a href="#" className="btn btn-dark-alt">
            LOGIN
          </a>
          <a href="#" className="btn btn-white-alt-hover">
            REGISTER
          </a>
        </div>
      </div>
    </div>
  );
};

export default Navbar;
