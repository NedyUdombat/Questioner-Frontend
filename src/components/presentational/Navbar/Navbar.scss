@import '../../../assets/sass/index';

* {
  box-sizing: border-box;
}

body {
  font-family: $raleway;
  margin: 0;
}

.nav {
  align-items: center;
  background-color: transparent;
  display: flex;
  height: 5rem;
  justify-content: space-between;
  position: relative;
  width: 100%;
}

.nav-dark {
  background-color: $primary-black;
  padding: 0 6%;
}

.nav > .nav-header {
  display: inline;
}

.nav > .nav-header > .nav-title {
  color: #fff;
  display: inline-block;
  font-family: $sacramento;
  font-size: 2.5rem;
}

.nav > .nav-btn {
  display: none;
}

.nav > .nav-links {
  display: flex;
  font-size: 1.125rem;
}

.nav > .nav-links > a {
  margin: 0 0 0 2rem;
  padding: 0 1.5rem;
  text-decoration: none;
}

.nav > .nav-links > a > .nav-link {
  color: $primary-white;
}

.nav > #nav-check {
  display: none;
}

@media (max-width: 37.5rem) {
  .nav > .nav-btn {
    display: inline-block;
    position: absolute;
    right: 3rem;
    top: .625rem;
  }

  .nav > .nav-btn > label {
    cursor: pointer;
    display: inline-block;
    height: 2.625rem;
    padding: .8125rem;
    width: 2.625rem;
  }

  .nav > .nav-btn > label:hover,
  .nav #nav-check:checked ~ .nav-btn > label {
    background-color: transparent;
  }

  .nav > .nav-links {
    background-color: $primary-white;
    display: block;
    left: 0;
    max-height: 300%;
    overflow: hidden;
    position: absolute;
    top: 5rem;
    transition: .5s ease-in;
    width: 100%;
  }

  .nav > .nav-links > a {
    margin-bottom: 1rem;
    width: 90%;
  }

  .nav > #nav-check:not(:checked) ~ .nav-links {
    height: 0;
  }

  .nav > #nav-check:checked ~ .nav-links {
    height: calc(100vh - 5rem);
    overflow-y: auto;
  }

  .nav-dark {
    background-color: $primary-black;
    padding: 0 6%;
  }
}
