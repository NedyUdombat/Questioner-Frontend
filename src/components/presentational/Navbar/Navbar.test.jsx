import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import { Navbar } from './Navbar';

const props = {
  username: jest.fn(),
  logOut: jest.fn(),
  className: 'navbar',
};

describe('Navbar', () => {
  it('should render correctly', () => {
    const component = mount(
      <BrowserRouter>
        <Navbar {...props} />
      </BrowserRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
