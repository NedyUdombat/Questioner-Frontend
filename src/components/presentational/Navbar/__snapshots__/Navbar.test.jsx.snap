// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
<Navbar>
  <div>
    <div
      className="nav"
    >
      <input
        id="nav-check"
        type="checkbox"
      />
      <div
        className="nav-header"
      >
        <p
          className="nav-title"
        >
          Questioner
        </p>
      </div>
      <div
        className="nav-btn"
      >
        <label
          htmlFor="nav-check"
        >
          <Icon
            as="i"
            color="grey"
            name="bars"
            size="big"
          >
            <i
              aria-hidden="true"
              className="grey bars big icon"
              onClick={[Function]}
            />
          </Icon>
        </label>
      </div>
      <div
        className="nav-links"
      >
        <a
          className="btn btn-transparent"
          href="#"
        >
          MEETUPS
        </a>
        <a
          className="btn btn-dark-alt"
          href="#"
        >
          LOGIN
        </a>
        <a
          className="btn btn-white-alt-hover"
          href="#"
        >
          REGISTER
        </a>
      </div>
    </div>
  </div>
</Navbar>
`;
